### PR TITLE
ipq40xx/generic: Install irqbalance to improve performance

### PR DESCRIPTION
**Pull request by robertfoss**
This PR supersedes #40 and fixes issue #67, which can be closed if this PR is merged.

### CODE REVIEW

File: group_vars/location_gubener/owm.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ---
 location_nice: Gubener37
-latitude: 52.510169
-longitude: 13.450146
+latitude: 52.50945371664022
+longitude: 13.447127850710801
```
Comment (PolynomialDivision) [inline group_vars/location_gubener/owm.yml:4]: What is happening here? Can you rebase?

Comment (PolynomialDivision) [reply group_vars/location_gubener/owm.yml:4]: The one commit should already be merged:
https://github.com/Freifunk-Spalter/bbb-configs/commit/6bc610d964b55cc9b14f44d909959b6fbb1679e1

**Comment by PolynomialDivision** [comment]
Are u sure you don't need any configs for this package?

**Comment by robertfoss** [comment]
I added some configuration for this package. But I'm not sure if it is correct.

**Comment by robertfoss** [comment]
As per @spolack's suggestion, I had a look at adding a multicore flag to `group_vars/target_XXX`. But I'm not sure where to read the flag and then dynamically add the `irqbalance` package.

**Comment by spolack** [comment]
> As per @spolack's suggestion, I had a look at adding a multicore flag to `group_vars/target_XXX`. But I'm not sure where to read the flag and then dynamically add the `irqbalance` package.


I'd propose to create a file, where we put all packages which are installed based on certain conditions. I am already working on other changes which would rely on that structure.

```sh
spolack@spolack:~/git/bbb-configs$ cat roles/cfg_openwrt/tasks/additional_packages.yml
- name: "Add airos-dfs-reset if nescessary"
  set_fact:
    packages: "{{ packages + ['airos-dfs-reset'] }}"
  when:
    - airos_dfs_reset is defined and airos_dfs_reset | length > 0
    - role == 'core_router'
```


`additional_packages.yml` can be then included in the main.yml:

https://github.com/Freifunk-Spalter/bbb-configs/blob/6bc610d964b55cc9b14f44d909959b6fbb1679e1/roles/cfg_openwrt/tasks/main.yml#L8

**Comment by robertfoss** [comment]
That looks rather neat to me. Can we rename the file to `conditional_packages.yml`?

**Comment by robertfoss** [comment]
I reworked this series to use the above mentioned conditional package strategy. 

```
TASK [cfg_openwrt : Add irqbalance if multicore] *******************
task path: /home/robertfoss/tmp/bbb-configs/roles/cfg_openwrt/tasks/conditional_packages.yml:2
ok: [gubener-core] => {
    "ansible_facts": {
        "packages": [
            "mtr",
            "tcpdump",
            "iperf3",
            "luci-ssl",
            "luci-app-statistics",
            "collectd-mod-uptime",
            "collectd-mod-memory",
            "collectd-mod-ping",
            "vnstat",
            "-ppp",
            "-ppp-mod-ppoe",
            "-wpad-basic",
            "-wpad-basic-wolfssl",
            "hostapd-wolfssl",
            "nftables",
            "kmod-nft-bridge",
            "-naywatch",
            "-kmod-ath10k-ct -ath10k-firmware-qca4019-ct",
            "kmod-ath10k ath10k-firmware-qca4019",
            "babeld",
            "luci-app-babeld",
            "collectd-mod-olsrd",
            "collectd-mod-exec",
            "collectd-mod-conntrack",
            "collectd-mod-snmp",
            "olsrd",
            "olsrd-mod-arprefresh",
            "olsrd-mod-dyn-gw",
            "olsrd-mod-jsoninfo",
            "olsrd-mod-nameservice",
            "olsrd-mod-txtinfo",
            "olsrd-utils",
            "luci-app-olsr",
            "luci-app-olsr-services",
            "luci-mod-falter",
            "falter-common",
            "luci-app-falter-owm",
            "luci-app-falter-owm-ant",
            "luci-app-falter-owm-cmd",
            "luci-app-falter-owm-gui",
            "iptables-mod-ipopt",
            "kmod-ipt-ipopt",
            "naywatch",
            "irqbalance"
        ]
    },
    "changed": false
}
```

As far as I can tell, it seems to be working.

**Pull request merged**